src/cudamatrix/cu-kernels.cu

```diff
@@ -754,6 +754,17 @@ static void _copy_from_vec_df(double* v_out, const Real* v_in, int dim) {
 }
 
 
+template<typename Real,typename OtherReal>
+__global__
+static void _copy_from_vec(OtherReal* v_out, const Real* v_in, int dim) {
+  int32_cuda i = blockIdx.x * blockDim.x + threadIdx.x;
+  //  if (blockIdx.y > 0) return;
+
+  if (i < dim) {
+    v_out[i] = (OtherReal) v_in[i];
```

Comment:
I looked this up, and CUDA does provide a way to control rounding in double->float casts. [See here](http://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__CAST.html#group__CUDA__MATH__INTRINSIC__CAST_1g64f62a06692dc6f683a2da3a25bd39de). Hopefully it does not matter here, though. We could get it if we need it using explicit template instantiation, I think.